### PR TITLE
Fix the check for gold check marks

### PR DESCRIPTION
**Pull request by jaredkotoff**
Fixes https://github.com/kheina-com/Blue-Blocker/issues/115

This PR increases the surface area of catching paid checkmarks by checking if the user is any of the "blockable" types and not depending on also having `is_blue_verified` flag set to true.

The initial assumption was the at gold check marks also had the `is_blue_verified` flag set to true, but it appears that not all gold/business checkmarks have this flag. 
See this comment for context and example of accounts that are not blocked at the moment: https://github.com/kheina-com/Blue-Blocker/issues/5#issuecomment-1569322041

example:
```json
{
    "result": {
        "__typename": "Tweet",
        "rest_id": "1642318156784979969",
        "core": {
            "user_results": {
                "result": {
                    "__typename": "User",
                    "id": "VXNlcjo4NzA2ODA5ODk=",
                    "rest_id": "870680989",
                    "affiliates_highlighted_label": {},
                    "has_graduated_access": true,
                    "is_blue_verified": false,
                    "profile_image_shape": "Square",
                    "legacy": {
                        "can_dm": true,
                        "can_media_tag": true,
                        "created_at": "Tue Oct 09 22:05:04 +0000 2012",
                        "default_profile": false,
                        "default_profile_image": false,
                        "description": "If you're reading this you need a beer. By following, you confirm you're 21+. Content & sharing for 21+ only. Enjoy Responsibly. https://t.co/VkFrH6PEgB",
                
                        "name": "Budweiser",
                        "normal_followers_count": 278766,
                        "pinned_tweet_ids_str": [
                            "1646958880654516226"
                        ],
                        "possibly_sensitive": false,
                        "profile_banner_url": "https://pbs.twimg.com/profile_banners/870680989/1675173257",
                        "profile_image_url_https": "https://pbs.twimg.com/profile_images/1620474706095644687/40hB2lKM_normal.jpg",
                        "profile_interstitial_type": "",
                        "screen_name": "budweiserusa",
                        "statuses_count": 74535,
                        "translator_type": "none",
                        "url": "https://t.co/d0nkBLfkOU",
                        "verified": false,
                        "verified_type": "Business",
                        "want_retweets": false,
                        "withheld_in_countries": []
                    },
                    "professional": {
                        "rest_id": "1570560815304175616",
                        "professional_type": "Business",
                        "category": [
                            {
                                "id": 250,
                                "name": "Brewery",
                                "icon_name": "IconBriefcaseStroke"
                            }
                        ]
                    }
                }
            }
        },
        "legacy": {
            "bookmark_count": 3,
            "bookmarked": false,
            "created_at": "Sun Apr 02 00:08:56 +0000 2023",
            "conversation_id_str": "1642318156784979969",
            "display_text_range": [
                0,
                15
            ],
            "entities": {
                "user_mentions": [],
                "urls": [],
                "hashtags": [],
                "symbols": []
            },
            "favorite_count": 515,
            "favorited": false,
            "full_text": "It’s beer time.",
            "is_quote_status": false,
            "lang": "en",
            "quote_count": 55,
            "reply_count": 2302,
            "retweet_count": 85,
            "retweeted": false,
            "user_id_str": "870680989",
            "id_str": "1642318156784979969"
        },
        "quick_promote_eligibility": {
            "eligibility": "IneligibleNotProfessional"
        }
    }
}
```

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -3,21 +3,21 @@ VERSION := $(shell cat build/manifest.json | jq .version)
 
 .PHONY: firefox
 firefox:
-	# ifneq (,$(wildcard blue-blocker-firefox-$(VERSION).zip))
-	# 	rm "blue-blocker-firefox-${VERSION}.zip"
-	# endif
+# ifneq (,$(wildcard blue-blocker-firefox-$(VERSION).zip))
```
Comment (jaredkotoff) [inline makefile:6]: Changes in this file are just so these lines aren't printed when ran. Indented comments are printed while un-indented ones are not.

File: src/shared.ts [hunk 2]
```diff
@@ -326,7 +328,9 @@ export function BlockBlueVerified(user: BlueBlockerUser, headers: any, config: C
         `did not block Twitter Blue verified user ${formattedUserName} because they have over a million followers and Elon is an idiot.`,
       );
     } else {
-      queueBlockUser(user, String(user.rest_id), headers, ReasonBlueVerified);
+      let reason = ReasonBlueVerified;
+      if (hasBlockableVerifiedTypes) reason = ReasonBusinessVerified;
```
Comment (MrAwesome) [inline src/shared.ts:332]: Nit - Generally I find this style of `if` more confusing and prefer the included {} for clarity, but definitely not worth blocking over

File: readme.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ npm run dev
 
 ### Firefox
 
-1. Run `make firefox` in the top right
+1. Run `make firefox`
```
Comment (MrAwesome) [inline readme.md:39]: Nice, I was about to open up a PR for this exact change

**Review by MrAwesome — APPROVED** [review]
Looks good to me! Still getting up to speed on the code but this seems pretty straightforward.

File: src/shared.ts (hunk 2 above)
Comment (jaredkotoff) [reply src/shared.ts:332]: for sure, I generally agree, too. Trying to break that habit.

**Pull request merged**